Source/Core/UICommon/NetPlayIndex.cpp

```diff
@@ -0,0 +1,297 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "UICommon/NetPlayIndex.h"
+
+#include <picojson/picojson.h>
+
+#include "Common/HttpRequest.h"
+#include "Common/Thread.h"
+#include "Common/Version.h"
+
+#include "Core/Config/NetplaySettings.h"
+
+NetPlayIndex::NetPlayIndex()
+{
+}
+
+NetPlayIndex::~NetPlayIndex()
+{
+  if (!m_secret.empty())
+    Remove();
+}
+
+static std::optional<picojson::value> ParseResponse(std::optional<std::vector<u8>> response)
```

Comment:
This function takes an optional vector but doesn't ever make sure there's any value. I see you already do the check before you call it so maybe it doesn't need to be optional?